Automattic/jetpack
# Decouple the disconnect flow from the dashboard component tree

This PR decouples the disconnection flow that is used in the Jetpack Dashboard so that it is rendered by the main app component. The goal of doing this is to be able to persist the disconnection flow modal after the site is disconnected in order to improve the UX of the disconnection process (showing disconnecting ... disconnected!) and to allow for some optional, anonymous feedback to be collected post-disconnection.

There is additional work to accomplish the end goal that will be handled in additional commits/ PRs. This PR only addresses changing how the disconnection modal is handled.

#### Changes proposed in this Pull Request:
* Adds new actions/ reducers to handle the disconnection modal state in the main app store
* Renders the disconnection modal in `main.jsx` 

#### Does this pull request change what data or activity we track or use?
No additional data is tracked or used in this PR at this time

#### Testing instructions:
* Go to the Jetpack dashboard on your test site 
* Click "Manage site Connection" to show the Disconnect Modal
* Confirm that you can still disconnect the site
